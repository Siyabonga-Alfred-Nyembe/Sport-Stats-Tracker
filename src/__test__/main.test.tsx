import { describe, it, expect, vi } from "vitest";

const mockRender = vi.fn();
const mockCreateRoot = vi.fn().mockReturnValue({ render: mockRender });

vi.mock("react-dom/client", () => ({
  createRoot: mockCreateRoot,
}));

vi.mock("../App.tsx", () => ({
  default: () => "MockedApp",
}));

describe("main.tsx", () => {
  it("mounts App inside StrictMode at #root", async () => {
    // Fake root element
    const fakeRoot = document.createElement("div");
    fakeRoot.id = "root";
    document.body.appendChild(fakeRoot);

    await import("../main.tsx");

    // Assertions
    expect(mockCreateRoot).toHaveBeenCalledWith(fakeRoot);
    expect(mockRender).toHaveBeenCalledTimes(1);
    expect(mockRender.mock.calls[0][0]).toMatchSnapshot();
  });
});
