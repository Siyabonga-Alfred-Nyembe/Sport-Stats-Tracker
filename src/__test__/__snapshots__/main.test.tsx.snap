// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`main.tsx > mounts App inside StrictMode at #root 1`] = `
<UNDEFINED>
  <default />
</UNDEFINED>
`;
